### PR TITLE
feat(hc): backfill user and team into NotificationSetting

### PR DESCRIPTION
**Pull request by vbro**
Step 3 of https://www.notion.so/sentry/Notification-Settings-4f60428198db41a68ac887ac8a89f324

### CODE REVIEW

File: src/sentry/migrations/0422_backfill_denormalize_notification_actor.py
Comment (mikejihbe) [inline src/sentry/migrations/0422_backfill_denormalize_notification_actor.py:36]: I think we'll need this to be True. Then we can open an ops ticket to run it once its merged

Comment (vbro) [reply src/sentry/migrations/0422_backfill_denormalize_notification_actor.py]: This will be run after https://github.com/getsentry/sentry/pull/47118 merges

File: src/sentry/migrations/0422_backfill_denormalize_notification_actor.py
Comment (wedamija) [inline src/sentry/migrations/0422_backfill_denormalize_notification_actor.py:22]: You can't use `.update` in migrations, this exists only in the real models, not the fake django migration versions of them.

Best to use `.save(update_fields=["team_id"])`

**Review by wedamija — COMMENTED** [review]
Migration lgtm, I'll leave it to your team to validate the logic

**Review by mikejihbe — APPROVED** [review]

**Review by dashed — APPROVED** [review]

**Pull request merged**
